packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -31,6 +35,106 @@ type BaseButtonWithIconProps = BaseButtonCommonProps & {
 
 export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
 
+const StyledBaseText = styled(BaseText)(
+  ({
+    hasIcon,
+    iconPosition,
+    iconSpacing,
+  }: BaseTextProps &
+    Pick<BaseButtonProps, 'iconPosition'> & { hasIcon: boolean; iconSpacing: string }) => `
+    padding-left: ${hasIcon && iconPosition === 'left' ? iconSpacing : 0};
+    padding-right: ${hasIcon && iconPosition === 'right' ? iconSpacing : 0};
+`,
+);
+
+type ConfigProps = {
```

Comment:
Here `prop` indicates that it is a prop that needs to be returned and passed to a DS component like `BaseText` whereas `raw value` indicates that it is a raw value that needs to be passed down to a `styled` component like `StyledBaseButton` to be consumed directly without computations. 

This is causing quite a confusion since `iconSpacing` will be a raw value like `0rem 1rem 0rem 1rem` but `iconColor` would be a prop value like `action.icon.primary.default`

Reply:
why do you need to explicitly have that distinction? the type suffice what we are doing right?

Reply:
It does. It's just annoying to have `iconColor` as a dot notation string vs `buttonColor`(see the next PR) as a direct string color value